### PR TITLE
Add ETW start and stop events for file changes

### PR DESCRIPTION
**Pull request by DrewScoggins**
This adds etw events for start and end on the TryChange methods. This is for both static and compilation changes.

### CODE REVIEW

File: src/BuiltInTools/dotnet-watch/HotReload/CompilationHandler.cs [hunk 1]
```diff
@@ -75,7 +75,7 @@ public async ValueTask<bool> TryHandleFileChange(DotNetWatchContext context, Fil
             {
                 return false;
             }
-
+
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/HotReload/CompilationHandler.cs:78]:
```suggestion
```

File: src/BuiltInTools/dotnet-watch/HotReload/HotReloadEventSource.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+using System;
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/HotReload/HotReloadEventSource.cs:1]:
```suggestion
// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System;
```

File: src/BuiltInTools/dotnet-watch/HotReload/HotReload.cs [hunk 3]
```diff
@@ -26,12 +26,20 @@ public async ValueTask InitializeAsync(DotNetWatchContext dotNetWatchContext, Ca
 
         public async ValueTask<bool> TryHandleFileChange(DotNetWatchContext context, FileItem file, CancellationToken cancellationToken)
         {
-            if (await _staticFileHandler.TryHandleFileChange(context, file, cancellationToken))
+            HotReladEventSource.Log.HotReloadStaticFileStart(file.FilePath);
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/HotReload/HotReload.cs:29]: Is it interesting to try and track each of these individually? Typically only one of these would work (it's either a css file and we tell the browser to refresh or it's a .cs / .razor file and we have to produce an EnC delta and apply it).

File: src/BuiltInTools/dotnet-watch/HotReload/StaticFileHandler.cs [hunk 4]
```diff
@@ -28,9 +28,10 @@ public async ValueTask<bool> TryHandleFileChange(DotNetWatchContext context, Fil
             {
                 return false;
             }
-
+            // Event for start to handle static file change
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/HotReload/StaticFileHandler.cs:31]: Undo?

File: src/BuiltInTools/dotnet-watch/HotReload/HotReloadEventSource.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        [Event(2, Message = "Hot reload finished for {0}", Level = EventLevel.Informational, Keywords = Keywords.Perf)]
+        public void HotReloadStaticFileEnd(string fileChanged) { WriteEvent(2, fileChanged); }
+
+        [Event(3, Message = "Hot reload started for {0}", Level = EventLevel.Informational, Keywords = Keywords.Perf)]
+        public void HotReloadCompilationStart(string fileChanged) { WriteEvent(3, fileChanged); }
+
+        [Event(4, Message = "Hot reload finished for {0}", Level = EventLevel.Informational, Keywords = Keywords.Perf)]
+        public void HotReloadCompilationEnd(string fileChanged) { WriteEvent(4, fileChanged); }
+
+        public static HotReladEventSource Log = new HotReladEventSource();
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/HotReload/HotReloadEventSource.cs:26]:
```suggestion
        public static readonly HotReladEventSource Log = new HotReladEventSource();
```

**Review by pranavkm — APPROVED** [review]

**Comment by DrewScoggins** [comment]
The test failure does not look related to my change

```
Running C:\h\w\B31E09B0\p\d\dotnet.exe msbuild /t:Build C:\h\w\B31E09B0\t\dotnetSdkTests\yl3hzewn.gk0\DesktopUsingP---33FC39E7\DesktopUsingPackageWithSatellites\DesktopUsingPackageWithSatellites.csproj /restore /v:normal
    Microsoft.NET.Build.Tests.GivenThatWeWantToBuildADesktopExe.It_places_package_satellites_correctly(crossTarget: True) [FAIL]
      Expected command to pass but it did not.
      File Name: C:\h\w\B31E09B0\p\d\dotnet.exe
Process ID: 6268
      Arguments: msbuild /t:Build C:\h\w\B31E09B0\t\dotnetSdkTests\yl3hzewn.gk0\DesktopUsingP---B0693BE4\DesktopUsingPackageWithSatellites_cross\DesktopUsingPackageWithSatellites_cross.csproj /restore /v:normal
      Exit Code: 1
      StdOut:
      Microsoft (R) Build Engine version 16.10.0-preview-21175-01+afd0b6210 for .NET
      Copyright (C) Microsoft Corporation. All rights reserved.
      
      Build started 3/26/2021 1:48:27 AM.
           1>Project "C:\h\w\B31E09B0\t\dotnetSdkTests\yl3hzewn.gk0\DesktopUsingP---B0693BE4\DesktopUsingPackageWithSatellites_cross\DesktopUsingPackageWithSatellites_cross.csproj" on node 1 (Restore target(s)).
           1>_GetAllRestoreProjectPathItems:
               Determining projects to restore...
             Restore:
               Restoring packages for C:\h\w\B31E09B0\t\dotnetSdkTests\yl3hzewn.gk0\DesktopUsingP---B0693BE4\DesktopUsingPackageWithSatellites_cross\DesktopUsingPackageWithSatellites_cross.csproj...
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/fluentvalidation/index.json
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/c9f8ac11-6bd8-4926-8306-f075241547f7/nuget/v3/flat2/fluentvalidation/index.json
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/flat2/fluentvalidation/index.json
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/fluentvalidation/index.json
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/d1622942-d16f-48e5-bc83-96f4539e7601/nuget/v3/flat2/fluentvalidation/index.json
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/f0702dc8-c92f-403c-a6b1-ae76b62614c8/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/516521bf-6417-457e-9a9c-0a4bdfde03e7/nuget/v3/flat2/fluentvalidation/index.json
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/fluentvalidation/index.json
                 GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/c9f8ac11-6bd8-4926-8306-f075241547f7/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/c9f8ac11-6bd8-4926-8306-f075241547f7/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/d1622942-d16f-48e5-bc83-96f4539e7601/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/d1622942-d16f-48e5-bc83-96f4539e7601/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/c9f8ac11-6bd8-4926-8306-f075241547f7/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/c9f8ac11-6bd8-4926-8306-f075241547f7/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/d1622942-d16f-48e5-bc83-96f4539e7601/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/d1622942-d16f-48e5-bc83-96f4539e7601/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/78665e4c-b767-412b-9804-2b1ef7a48b8a/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/f0702dc8-c92f-403c-a6b1-ae76b62614c8/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/f0702dc8-c92f-403c-a6b1-ae76b62614c8/nuget/v3/flat2/fluentvalidation/index.json
               Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/516521bf-6417-457e-9a9c-0a4bdfde03e7/nuget/v3/flat2/fluentvalidation/index.json'.
               No such host is known. (pkgs.dev.azure.com:443)
                 No such host is known.
                 GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/516521bf-6417-457e-9a9c-0a4bdfde03e7/nuget/v3/flat2/fluentvalidation/index.json
           1>C:\h\w\B31E09B0\p\d\sdk\6.0.100-ci\NuGet.targets(131,5): error : Failed to retrieve information about 'FluentValidation' from remote source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/c9f8ac11-6bd8-4926-8306-f075241547f7/nuget/v3/flat2/fluentvalidation/index.json'. [C:\h\w\B31E09B0\t\dotnetSdkTests\yl3hzewn.gk0\DesktopUsingP---B0693BE4\DesktopUsingPackageWithSatellites_cross\DesktopUsingPackageWithSatellites_cross.csproj]
      C:\h\w\B31E09B0\p\d\sdk\6.0.100-ci\NuGet.targets(131,5): error :   No such host is known. (pkgs.dev.azure.com:443) [C:\h\w\B31E09B0\t\dotnetSdkTests\yl3hzewn.gk0\DesktopUsingP---B0693BE4\DesktopUsingPackageWithSatellites_cross\DesktopUsingPackageWithSatellites_cross.csproj]
      C:\h\w\B31E09B0\p\d\sdk\6.0.100-ci\NuGet.targets(131,5): error :   No such host is known. [C:\h\w\B31E09B0\t\dotnetSdkTests\yl3hzewn.gk0\DesktopUsingP---B0693BE4\DesktopUsingPackageWithSatellites_cross\DesktopUsingPackageWithSatellites_cross.csproj]
```

**Comment by pranavkm** [comment]
Unrelated. AzDo feeds throttle if there's too many builds at a time and that's the error you would see. Just retry the build when it becomes ready.

**Review by brianrob — APPROVED** [review]

**Pull request merged**